pkgs/by-name/sn/snowflake-cli/package.nix

```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  installShellFiles,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "snowflake-cli";
+  version = "3.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "snowflakedb";
+    repo = "snowflake-cli";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-QBYdzKKkakgwCD12bN7CerjfvMnxvt2cJ0NqTIjHkWw=";
+  };
+
+  build-system =
+    (with python3Packages; [
+      hatch-vcs
+      hatchling
+      pip
+    ])
+    ++ [ installShellFiles ];
+
+  dependencies = with python3Packages; [
+    jinja2
+    pluggy
+    pyyaml
+    rich
+    requests
+    requirements-parser
+    setuptools
+    tomlkit
+    typer
+    urllib3
+    gitpython
+    pydantic
+    snowflake-connector-python
+  ];
+
+  nativeCheckInputs = with python3Packages; [
+    pytestCheckHook
+    syrupy
+    coverage
+    pytest-randomly
+  ];
+
+  pytestFlagsArray = [
+    "--snapshot-warn-unused" # Turn unused snapshots into a warning and not a failure
+  ];
+
+  disabledTests = [
+    "integration"
+    "spcs"
+    "loaded_modules"
+    "integration_experimental"
+    "test_snow_typer_help_sanitization" # Snapshot needs update?
+    "test_help_message" # Snapshot needs update?
+    "test_executing_command_sends_telemetry_usage_data" # Fails on mocked version
+  ];
+
+  pythonRelaxDeps = true;
+
+  postInstall = ''
+    export HOME=$(mktemp -d)
+
+    # Typer tries to guess the current shell by default
+    export _TYPER_COMPLETE_TEST_DISABLE_SHELL_DETECTION=1
+    installShellCompletion --cmd snow \
```

Comment:
This will not work when cross-compiling #308283

Reply:
Thanks, I was not aware of this. I added the conditional from [this PR](https://github.com/NixOS/nixpkgs/pull/353430/files).